tests/utils/test_mlir_lexer.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+
+from xdsl.utils.lexer import Input
+from xdsl.utils.mlir_lexer import StringLiteral
+
+
+def _create_string_literal(input_str: str) -> StringLiteral:
+    """Create a StringLiteral from a given input string."""
+    input_data = Input(input_str, "test.mlir")
+    return StringLiteral(0, len(input_str), input_data)
```

Comment:
since this is used only in the test below, let's inline it? It'll simplify the `parametrize` call which I'm finding a bit hard to read

Reply:
Done by https://github.com/xdslproject/xdsl/pull/4834/commits/dcf7f34756732fa8b42b66f37cb09cafe183bee8